Refuse to start a print if power is lost before arming uvlo

wavexx opened the pull request:
If power has been lost during startup already a falling edge would be
skipped, causing the print to continue and lose its state without
being able to save again.

Check for a low line after arming the interrupt and simply wait
for reset.

PFW-1079

wavexx committed:
Refuse to start a print if power is lost before arming uvlo

If power has been lost during startup already a falling edge would be
skipped, causing the print to continue and lose its state without
being able to save again.

Check for a low line after arming the interrupt and simply wait
for reset.

leptun commented:
Hmmm... checking for the low pin might not be ideal. There was another issue where they asked for checking for shorted PowerPanic pin during selftest. This would make doing that impossible without some changes. What if we do that only if EEPROM_UVLO is different than 0? This way it would also allow me to use manual PowerPanic on my MK2.5SS that doesn’t have the hardware for this function.

leptun commented:
@wavexx Similar as you wrote in #1787, except we also check for EEPROM_UVLO.

wavexx committed:
Only trigger a quick reset if there's a pending saved state

Thanks to @leptun

wavexx committed:
Display "POWER PANIC DETECTED" when possible

Both during early init and in uvlo_tiny, display "POWER PANIC DETECTED"
if enough charge is left.

This is not worth doing in regular uvlo_, as we want to give full
priority to the X motor
